src/Authenticators/LDAPLostPasswordHandler.php

```diff
@@ -0,0 +1,238 @@
+            return $controller->redirect('lostpassword');
+        }
+
+        if ($member) {
+            /** @see MemberLoginForm::forgotPassword */
+            $token = $member->generateAutologinTokenAndStoreHash();
+            $e = Email::create()
+                ->setSubject(
+                    _t(
+                        'Silverstripe\\Security\\Member.SUBJECTPASSWORDRESET',
+                        'Your password reset link',
+                        'Email subject'
+                    )
+                )
+                ->setHTMLTemplate('SilverStripe\\Control\\Email\\ForgotPasswordEmail')
+                ->setData($member)
+                ->setData(['PasswordResetLink' => Security::getPasswordResetLink($member, $token)]);
+            $e->setTo($member->Email);
+            $e->send();
+            return $controller->redirect($controller->Link('passwordsent/') . urlencode($data['Login']));
+        } elseif ($data['Login']) {
+            // Avoid information disclosure by displaying the same status,
+            // regardless whether the email address actually exists
+            return $controller->redirect($controller->Link('passwordsent/') . urlencode($data['Login']));
+        } else {
+            if (Config::inst()->get(LDAPAuthenticator::class, 'allow_email_login') === 'yes') {
+                $form->sessionMessage(
+                    _t(
+                        'SilverStripe\\ActiveDirectory\\Forms\\LDAPLoginForm.ENTERUSERNAMEOREMAIL',
+                        'Please enter your username or your email address to get a password reset link.'
+                    ),
+                    'bad'
+                );
+            } else {
+                $form->sessionMessage(
+                    _t(
+                        'SilverStripe\\ActiveDirectory\\Forms\\LDAPLoginForm.ENTERUSERNAME',
+                        'Please enter your username to get a password reset link.'
+                    ),
+                    'bad'
+                );
+            }
+            return $controller->redirect($controller->Link('lostpassword'));
+        }
+    }
+
+    /**
+     * Factory method for the lost password form
+     *
+     * @return Form Returns the lost password form
+     */
+    public function lostPasswordForm()
+    {
+        if (Config::inst()->get(LDAPAuthenticator::class, 'allow_email_login') === 'yes') {
+            $loginField = TextField::create(
+                'Login',
+                _t('SilverStripe\\ActiveDirectory\\Forms\\LDAPLoginForm.USERNAMEOREMAIL', 'Username or email'),
+                null,
+                null,
+                $this
+            );
+        } else {
+            $loginField = TextField::create(
+                'Login',
+                _t(
+                    'SilverStripe\\ActiveDirectory\\Forms\\LDAPLoginForm.USERNAME',
+                    'Username'
+                ),
+                null,
+                null,
+                $this
```

Comment:
I reckon we could consolidate this a little by only passing the first two arguments in each condition above, then below add `$loginField->setForm($this);`